packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -1372,6 +1370,55 @@ class SemanticsNode extends AbstractNode with DiagnosticableTreeMixin {
     _dirty = false;
   }
 
+  /// Builds a new list made of [_children] sorted in semantic traversal order.
+  List<SemanticsNode> _childrenInTraversalOrder() {
+    final List<SemanticsNode> childrenInDefaultOrder = _childrenInDefaultOrder(_children);
+
+    // List.sort does not guarantee stable sort order. Therefore, children are
+    // first partitioned into groups that have compatible sort keys, i.e. keys
+    // in the same group can be compared to each other. These groups stay in
+    // the same place. Only children within the same group are sorted.
+    final List<_TraversalSortNode> everythingSorted = <_TraversalSortNode>[];
+    final List<_TraversalSortNode> sortNodes = <_TraversalSortNode>[];
+    SemanticsSortKey lastSortKey;
+    for (int position = 0; position < childrenInDefaultOrder.length; position += 1) {
+      final SemanticsNode child = childrenInDefaultOrder[position];
+      final SemanticsSortKey sortKey = child.sortKey;
+      lastSortKey = position > 0
+          ? childrenInDefaultOrder[position - 1].sortKey
+          : null;
+      final bool isCompatibleWithPreviousSortKey = position == 0 ||
+          sortKey.runtimeType == lastSortKey.runtimeType &&
+          (sortKey == null || sortKey.name == lastSortKey.name);
+      if (!isCompatibleWithPreviousSortKey && sortNodes.isNotEmpty) {
+        // Do not sort groups with null sort keys. List.sort does not guarantee
+        // a stable sort order.
```

Comment:
Correct. I do not expect this to be a problem though. If you are specifying explicit keys, it does not make sense to supply equal keys.

Reply:
Actually, because we are sorting by `_TraversalSortNode` there's no way you can have this situation. `_TraversalSortNode` uses list position in paint order as the tie breaker, so there's no way you can get two equal `_TraversalSortNode`s.